scripts/automation/Radius/Functions/Public/Get-JCRGlobalVars.ps1

```diff
@@ -0,0 +1,182 @@
+                    # create the hashtable:
+                    $userAssociationList = New-Object System.Collections.Hashtable
+                    foreach ($item in $reportContent) {
+                        if ($item.user_object_id -And $item.resource_object_id) {
+                            if (-not $userAssociationList[$item.user_object_id]) {
+                                $userAssociationList.add(
+                                    $item.user_object_id, @{
+                                        'systemAssociations' = @($item | Select-Object -Property @{Name = 'systemId'; Expression = { $_.resource_object_id } }, hostname, @{Name = 'osFamily'; Expression = { $_.device_os } });
+                                        'userData'           = @($item | Select-Object -Property email, username)
+                                    }) | Out-Null
+                            } else {
+                                $userAssociationList[$item.user_object_id].systemAssociations += @($item | Select-Object -Property @{Name = 'systemId'; Expression = { $_.resource_object_id } }, hostname, @{Name = 'osFamily'; Expression = { $_.device_os } })
+                            }
+                        }
+                    }
+                    # write out the association hash
+                    $userAssociationList | ConvertTo-Json -Depth 10 |  Out-File "$JCScriptRoot/data/associationHash.json"
+                } else {
+                    $userAssociationList = Get-Content -Raw -Path "$JCScriptRoot/data/associationHash.json" | ConvertFrom-Json -Depth 6 -AsHashtable
+
+                }
+                # finally write out the data to file:
+                $users | ConvertTo-Json -Depth 100 -Compress |  Out-File "$JCScriptRoot/data/userHash.json"
+                $systems | ConvertTo-Json -Depth 10 |  Out-File "$JCScriptRoot/data/systemHash.json"
+                $radiusMemberList | ConvertTo-Json |  Out-File "$JCScriptRoot/data/radiusMembers.json"
+            }
+            $false {
+                # write-host "It's been $($lastUpdateTimespan.hours) hours since we last pulled user, system and association data, no need to update"
+                $userAssociationList = Get-Content -Raw -Path "$JCScriptRoot/data/associationHash.json" | ConvertFrom-Json -Depth 6 -AsHashtable
+            }
+        }
+    }
+    end {
+        switch ($update) {
+            $true {
+                # set global vars
+                $Global:JCRUsers = $users
+                $Global:JCRSystems = $systems
+                $Global:JCRAssociations = $userAssociationList
+                $Global:JCRRadiusMembers = $radiusMemberList
+                # update the settings date
+                Set-JCRSettingsFile -globalVarslastUpdate (Get-Date)
+                # update users.json
+                Update-JCRUsersJson
+            }
+            $false {
+                # set global vars from local cache
```

Comment:
In daeca953583f2fbd5960ff586716c01d7e8e5794 I updated each of the global variables with a namespace `$JCR_SOME_VARIABLE` to prevent this from interfering with other variables